### DISCUSSION TITLE
Module NativeFileSelector: Object property '[object object ]' cannot have type 'FunctionTypeAnnotation'.

### DISCUSSION THREAD

**Discussion by dairan915** [reactwg/react-native-new-architecture#219] [Q&A]
![image](https://github.com/user-attachments/assets/0bd35e51-3f7b-43e1-ac56-009e462a77e5)
![image](https://github.com/user-attachments/assets/ac1d754a-ed4d-4972-9237-8aab052ca8a3)
When SelectOptions has onDon and onCancel properties, codegen fails, but the interface only passes fixly one parameter of type SelecOptions. Therefore, onDone and onCancel cannot be passed as parameters. how to deal with this problem in the case of a new architecture?

**Comment by cortinico** [comment]
Can you please share your code as text and not as images, as it makes it really hard to reproduce and it's terrible for searchability/SEO.
